core/server.go

```diff
@@ -5,21 +5,102 @@ package core
 import (
 	"context"
 
+	"github.com/anz-bank/sysl-go/config"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/sirupsen/logrus"
 )
 
+type ServerParams struct {
+	Ctx                context.Context
+	Name               string
+	Config             *config.DefaultConfig
+	logger             *logrus.Logger
+	restManager        RestManager
+	grpcManager        GrpcManager
+	prometheusRegistry *prometheus.Registry
+}
+
+func NewServerParams(ctx context.Context, name string, config *config.DefaultConfig) *ServerParams {
```

Comment:
Can the `NewServerParams` function signature become:
```go
func NewServerParams(ctx context.Context, name string, opts ...ServerOption) error
```

and the `Start` function become:
```go
func (params *ServerParams) Start() error
```

The `config.DefaultConfig` isn't required to start the server.